docs/registry/create/_index.md

```diff
@@ -535,18 +528,59 @@ func (b *myBase) Close(ctx context.Context) error {
+For example, the `SetPower()` implementation in the example code above returns a `multierr` value (as provided by the [`multierr` package](https://pkg.go.dev/go.uber.org/multierr)), which allows for transparently combining multiple Go `error` return values together.
+This matches the `error` return type of the built-in `SetPower()` method as defined in the Viam Go SDK in the file [`base.go`](https://github.com/viamrobotics/rdk/blob/main/components/base/base.go).
+
+For more information on the base component API methods used in this example, see the following resources:
+
+- [Go SDK documentation for the `base` package](https://pkg.go.dev/go.viam.com/rdk/components/base#pkg-functions)
+- [Base API methods](https://docs.viam.com/components/base/#api)
+
+You can find additional Go example modules in the [Go SDK `examples` directory](https://github.com/viamrobotics/rdk/blob/main/examples/).
 
 {{% /tab %}}
 {{% tab name="C++" %}}
 
-Your new resource model must implement all the methods that the Viam RDK requires, and should match the built-in {{< glossary_tooltip term_id="subtype" text="subtype" >}} API, like [`rdk:component:base`](https://cpp.viam.dev/classviam_1_1sdk_1_1Base.html).
+First, inspect the built-in class provided by the resource API that you are extending.
+
+For example, if you wanted to add support for a new [base component](/components/base/) to Viam, you would start by looking at the built-in base component class.
+The built-in `Base` class is defined in the [Viam C++ SDK](https://cpp.viam.dev/) in the following files:
+
+<!-- prettier-ignore -->
+| Resource Model File | Description |
+| ------------------- | ----------- |
+| [components/base/base.cpp](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/base/base.cpp) | Defines the specific functionality of the built-in `Base` class. |
+| [components/base/base.hpp](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/base/base.hpp) | Defines the implementation of functionality defined in `base.cpp`, which includes the declaration of several built-in functions such as `move_straight()`. |
+
+{{% alert title="Tip" color="tip" %}}
+You can view the other built-in component classes in similar fashion.
+For example, the `Camera` class is defined in [camera.cpp](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/camera/camera.cpp) and its functions declared in [camera.hpp](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/camera/camera.hpp), while the `Sensor` class is defined in [sensor.cpp](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/sensor/sensor.cpp) and its functions declared in [sensor.hpp](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/sensor/sensor.hpp).
```

Comment:
(minor) I think it would help to be slightly more explicit about the fact that resource base classes are all abstract, and the resource-y methods (`stop()`, `get_image()`, etc.) are not defined in, e.g., `camera.cpp`.